metaflow/metaflow_current.py

```diff
@@ -62,7 +62,12 @@ def _set_env(
 
     def _update_env(self, env):
         for k, v in env.items():
-            setattr(self.__class__, k, property(fget=lambda _, v=v: v))
+            if isinstance(v, property):
+                setattr(self.__class__, k, v)
+            elif callable(v):
+                setattr(self, k, functools.partial(v, self))
+            else:
+                setattr(self.__class__, k, property(fget=lambda _, v=v: v))
 
     def __contains__(self, key: str):
```

Comment:
With the change above, I don't know how we want to treat __contains__ and get. When things were just properties, it was kind of an easy choice but with callables not 100% sure. Do we need the callables or can we just stick to adding properties?